src/engine/Engine.ts

```diff
@@ -81,32 +58,25 @@ export enum ScrollPreventionMode {
   All
 }
 
-/**
- * Interface describing the absolute CSS position of the game window. For use when [[DisplayMode.Position]]
- * is specified and when the user wants to define exact pixel spacing of the window.
- * When a number is given, the value is interpreted as pixels
- */
-export interface AbsolutePosition {
-  top?: number | string;
-  left?: number | string;
-  right?: number | string;
-  bottom?: number | string;
-}
-
 /**
  * Defines the available options to configure the Excalibur engine at constructor time.
  */
 export interface EngineOptions {
   /**
-   * Optionally configure the native canvas width of the game
+   * Optionally configure the native logical pixel width of the game (resolution)
    */
   width?: number;
 
   /**
-   * Optionally configure the native canvas height of the game
+   * Optionally configure the native logical pixel height of the game (resolution)
    */
   height?: number;
 
+  /**
+   * Optionally specify the size the viewport should take up onscreen
+   */
+  viewport?: { width: number; height: number };
```

Comment:
My only concern is that I think this represents a change in the way `width` and `height` is interpreted from previous versions. So, if someone had set their WxH to 1024x768, uptaking this will "work" but technically the meaning of `width` and `height` is different than what they had previously thought.

I wonder if it would be a smoother upgrade path to flip this. Instead of `width` and `height` being resolution, maybe `viewport` should change to `resolution`.

That way, using my example, if I had code like this:

```js
const game = new Engine({
  width: 1024,
  height: 768
});
```

Then after uptaking this next version, I could adjust the resolution of the game using:

```js
const game = new Engine({
  width: 1024,
  height: 768,
  resolution: Resolution.Gameboy
});
```

It also works well with the new `Resolution` presets.

Thoughts?

Reply:
We spoke offline, this makes a lot of sense.